include/swift/AST/DiagnosticsParse.def

```diff
@@ -707,6 +707,10 @@ ERROR(expected_close_to_config_stmt,stmt_parsing,none,
 ERROR(expected_close_after_else,stmt_parsing,none,
       "further conditions after #else are unreachable", ())
 
+/// Associatedtype Statement
+WARNING(use_associatedtype_keyword,stmt_parsing,none,
+        "typealias for associated types in a protocol will be removed in a future version of Swift; use the associatedtype keyword", ())
```

Comment:
Can we tighten this wording up a bit, and quote the keywords? Perhaps:

  "use of 'typealias' to declare associated types is deprecated; use 'associatedtype' instead"

?